### DISCUSSION TITLE
Solax X3 Hybrid Raspberry RS485 USB -> Unknown error occurred

### DISCUSSION THREAD

**Discussion by Mike5229** [wills106/homeassistant-solax-modbus#988] [Q&A]
Hello,

I just can't get the connection via Modbus to work.

The following setup:

Raspberry 4

OS: Dietpi

RS485 to USB adapter: https://www.amazon.de/Greluma-...B09P8CX69G?tag=pvforum-21

Network cable from COM to USB converter. Blue to A and blue/white to B

Solax X3 15.0 Firmware: 43 09 42

I use Home Assistant with HACS and the addon from Wills: https://github.com/wills106/homeassistant-solax-modbus

So far everything has worked, but when I start the integration in Home Assistant and make the Modbus setting, I always get a meaningless error message. (See pictures)

I have set the baud rate to 9600 on both the Solax and the Raspi as a test. The error remains.

Modbus channel is definitely 1

The USB stick is mounted by the Raspi as /dev/ttyUSB0

What am I doing wrong?

![2024-08-08 09_32_27-Window](https://github.com/user-attachments/assets/49d03979-3d8a-42ae-b5b9-e7f00634aa25)
![2024-08-08 09_32_04-Window](https://github.com/user-attachments/assets/b4111417-b21f-4652-a187-d01c2fe9f660)

**Comment by helge26** [comment]
Same problem here. Removed entity before update as suggested. When the error occurs, removed integration and reinstalled, but error remains.

The log is about

```
File "/config/custom_components/solax_modbus/config_flow.py", line 193, in _next_step_battery
    if user_input["support-battery"]:
       ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
KeyError: 'support-battery'
```

My environment:
- QCells hybrid inverter (Solax compatible)
- USB-RS485-adaptor from BerryBase.de as device /dev/serial/by-id/usb-1a86_USB2.0-Ser_-if00-port0
- Raspberry 4B 8GB with Home Assistant OS 12.4
- Home Assistant Core 2024.08

Helge

**Comment by vaskevics** [comment]
Can confirm that. Was working just fine before update. Tried to go back to 2024.08.4 but still the same error. Probably, if I did not remove the device before update, it would still work.

**Comment by ruimmsimoes** [comment]
I’m having the same issue when adding a SolaX inverter after upgrading to 2024.08.08

**Comment by butvoo** [comment]
I have the same problem, 
1. I have NOT removed Solax inverter from Solax Modbus integration :-( 
2.  upgraded Solax Modbus to 2024.8.8
3. I am not able to add Solax inverter, it error "Unknow error occurred"

Could you please help with break-fix to onboard Solax inverter back? 
meantime before the fix is applied, is it possible to downgrade to a working previous version? 

![image](https://github.com/user-attachments/assets/34b96a74-05c9-466b-933f-d8376910a4ca)

![image](https://github.com/user-attachments/assets/6e1de369-1302-4ee7-b80f-a29cf6e7a148)

**Comment by butvoo** [comment]
Workaround that works 👍  is 
1. add (fake) integration as TCP/ Ethernet
![image](https://github.com/user-attachments/assets/adfee2b9-2a70-4b25-81da-4832d68f112c)

2.  any temporary IP address can be used:
![image](https://github.com/user-attachments/assets/1964aa8e-977a-41bb-9397-547adb8857ca)

3. inside integration > 3dots > "Configure" > Change to "Serial" > Submit> Serial port name> Baudrate > Submit> ShutDown or Restart HA

![image](https://github.com/user-attachments/assets/46ac2cbd-6b78-4323-92f3-f24c1ced71cb)
![image](https://github.com/user-attachments/assets/2571f9d2-828b-43c8-a991-17b4f80ccabf)
![image](https://github.com/user-attachments/assets/639137ec-f80b-4159-90e2-558c31507b6c)

After restarting it loads Solax with the use of Serial
![image](https://github.com/user-attachments/assets/f5840c72-a0c9-4d94-ac23-186544f59fbb)

**Comment by wills106** [comment]
I have released [2024.08.11](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.08.11) which fixes an error with Serial connection.

**Reply by vaskevics** [reply]
Thank you for your effort in investigating and patching! Can confirm that the problem is gone. Removed entity, installed update and added inverter back and things are working just fine. 👍👍👍

**Comment by Mike5229** [comment]
The Problem ist solved with Version 2024.08.12!
Thanks!!!

**Reply by marcin85** [reply]
problem is still here..
